src/classes/CountryStateDistrict.cls

```diff
@@ -0,0 +1,115 @@
+            mapOfStatewithCountry.put(states.Country_Name__r.Name,mapOfDistrictsWithState);
+        }
+        //Displaying country ,state and district combination where start with s
+        system.debug('Start with'+ch+'all countries,state and districts'+mapOfStatewithCountry);
+        //Find such a districts which length is more or equal to some number
+        List<District__c> districts=[SELECT Name FROM District__c];
+        for(District__c district:  districts){
+            if(length<=district.Name.length() ){
+                system.debug(district.Name);
+            }
+        }
+    }
+    // Calculate Total number of States And Districts using Aggregate Queries
+    List<Sobject> sobjectListForDml=new List<Sobject> ();
+    public void totalNoOfStateAndDistrict(){
+        List<AggregateResult> stateAndDistrictAggregateResult=[SELECT Country_Name__c country_Id, COUNT(Id) Total_No_States,
+                                                               SUM(Total_Number_of_Districts__c)Total_No_Districts FROM State__c GROUP BY Country_Name__c]; 
+        for(AggregateResult result : stateAndDistrictAggregateResult){
+            Country__c obj= new Country__c();
+            obj.Total_Number_of_States__c=(Integer)result.get('Total_No_States');
+            obj.Total_Number_of_Districts__c=(Decimal)result.get('Total_No_Districts');
+            obj.Id=(Id)result.get('country_Id');
+            sobjectListForDml.add(obj);
+        }
+        if(!sobjectListForDml.isEmpty()){
+            update sobjectListForDml;
+        }
+    }
+    // Calculate Total number of Districts using Aggregate Queries
+    public void totalNoOfDistrict(){
+        
+        List<AggregateResult> DistrictAggregateResult=[SELECT State_Name__c state_Id,COUNT(Id)   
+                                                       Total_No_Districts FROM District__c GROUP BY State_Name__c];
+        for(AggregateResult result : DistrictAggregateResult){
+            State__c obj=new State__c();
+            obj.Total_Number_of_Districts__c=(Integer)result.get('Total_No_Districts');
+            obj.Id=(Id)result.get('state_Id');
+            sobjectListForDml.add(obj); 
+        }
+        if(!sobjectListForDml.isEmpty()){
+            update sobjectListForDml;
+        }
+    }
+    //CountryJSON Generator  method 
+    public void  countryJSONGenerator(string countryName){
+        List<Country__c> countryStateDistrictList=[SELECT Id,Name,Country_Code__c,Total_Number_of_States__c, 
+                                                   Total_Number_of_Districts__c,(SELECT Id,Name,State_Code__c,Total_Number_of_Districts__c FROM States__r) FROM Country__c
+                                                   WHERE Name = : countryName];
+        
+        if(!countryStateDistrictList.isEmpty()){
+            Country wrapperCountryObj= new Country();
+            wrapperCountryObj.countryName=countryStateDistrictList[0].Name;
```

Comment:
All this data can be passed to a Parametric constructor for the country wrapper class. and values can be set over there.